packages/domain-lib/test/unit/index.test.ts

```diff
@@ -1918,4 +1913,139 @@ describe("Settlements BC [Domain] - Unit Tests", () => {
+		expect(matrix1!.balancesByCurrency[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByCurrency[0].creditBalance).toEqual('200');
+
+		// Balances by state and currency:
+		expect(matrix1!.balancesByStateAndCurrency.length).toEqual(1);
+		expect(matrix1!.balancesByStateAndCurrency[0].state).toEqual('OPEN');
+		expect(matrix1!.balancesByStateAndCurrency[0].currencyCode).toEqual('EUR');
+		expect(matrix1!.balancesByStateAndCurrency[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByStateAndCurrency[0].creditBalance).toEqual('200');
+
+		// Balances by participant:
+		expect(matrix1!.balancesByParticipant.length).toEqual(2);
+		expect(matrix1!.balancesByParticipant[0].state).toEqual('OPEN');
+		expect(matrix1!.balancesByParticipant[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByParticipant[0].creditBalance).toEqual('0');
+		expect(matrix1!.balancesByParticipant[1].state).toEqual('OPEN');
+		expect(matrix1!.balancesByParticipant[1].debitBalance).toEqual('0');
+		expect(matrix1!.balancesByParticipant[1].creditBalance).toEqual('200');
+
+		const batchId_3: string = await aggregate.handleTransfer(reqTransferDto_3);
+		expect(batchId_1).toBeDefined();
+		expect(batchId_2).toEqual(batchId_1);
+		expect(batchId_3).toEqual(batchId_1);
+
+		// Fetch the matrix, which is not taking into account the 3rd transfer:
+		matrix1 = await settleMatrixReqRepo.getMatrixById(matrixId_1);
+		expect(matrix1).toBeDefined();
+		expect(matrix1!.id).toEqual(matrixId_1);
+		expect(matrix1!.state).toEqual('IDLE');
+
+		// balances by currency:
+		expect(matrix1!.balancesByCurrency.length).toEqual(1);
+		expect(matrix1!.balancesByCurrency[0].currencyCode).toEqual('EUR');
+		expect(matrix1!.balancesByCurrency[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByCurrency[0].creditBalance).toEqual('200');
+
+		// balances by state and currency:
+		expect(matrix1!.balancesByStateAndCurrency.length).toEqual(1);
+		expect(matrix1!.balancesByStateAndCurrency[0].state).toEqual('OPEN');
+		expect(matrix1!.balancesByStateAndCurrency[0].currencyCode).toEqual('EUR');
+		expect(matrix1!.balancesByStateAndCurrency[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByStateAndCurrency[0].creditBalance).toEqual('200');
+
+		// balances by participant:
+		expect(matrix1!.balancesByParticipant.length).toEqual(2);
+		expect(matrix1!.balancesByParticipant[0].state).toEqual('OPEN');
+		expect(matrix1!.balancesByParticipant[0].debitBalance).toEqual('200');
+		expect(matrix1!.balancesByParticipant[0].creditBalance).toEqual('0');
+		expect(matrix1!.balancesByParticipant[1].state).toEqual('OPEN');
+		expect(matrix1!.balancesByParticipant[1].debitBalance).toEqual('0');
+		expect(matrix1!.balancesByParticipant[1].creditBalance).toEqual('200');
+
+		// We need to re-calculate in order to account for the 3rd transfer, otherwise we have a transfer missing:
+		await aggregate.lockSettlementMatrixForAwaitingSettlement(matrixId_1);
```

Comment:
@myo-min-htet if the following two lines are removed, we have a stale matrix after the 3rd transfer. Which re-creates the scenario you are facing.
```nodejs
await aggregate.lockSettlementMatrixForAwaitingSettlement(matrixId_1);
await aggregate.settleSettlementMatrix(matrixId_1);
```